sqlglot/diff.py

```diff
@@ -0,0 +1,300 @@
+        target_args = [id(e) for e in _expression_only_args(target)]
+
+        args_lcs = set(
+            _lcs(source_args, target_args, lambda l, r: (l, r) in matching_set)
+        )
+
+        move_edits = []
+        for a in source_args:
+            if a not in args_lcs and a not in self._unmatched_source_nodes:
+                move_edits.append(Move(self._source_index[a]))
+
+        return move_edits
+
+    def _compute_matching_set(self):
+        leaves_matching_set = self._compute_leaf_matching_set()
+        matching_set = leaves_matching_set.copy()
+
+        ordered_unmatched_source_nodes = {
+            id(n[0]): None
+            for n in self._source.bfs()
+            if id(n[0]) in self._unmatched_source_nodes
+        }
+        ordered_unmatched_target_nodes = {
+            id(n[0]): None
+            for n in self._target.bfs()
+            if id(n[0]) in self._unmatched_target_nodes
+        }
+
+        for source_node_id in ordered_unmatched_source_nodes.keys():
+            for target_node_id in ordered_unmatched_target_nodes.keys():
+                source_node = self._source_index[source_node_id]
+                target_node = self._target_index[target_node_id]
+                if _is_same_label(source_node, target_node):
+                    similarity_score = _dice_coefficient(
+                        source_node.sql(), target_node.sql()
+                    )
+
+                    source_leaf_ids = {id(l) for l in _get_leaves(source_node)}
+                    target_leaf_ids = {id(l) for l in _get_leaves(target_node)}
+                    common_leaves_num = sum(
+                        1 if s in source_leaf_ids and t in target_leaf_ids else 0
+                        for s, t in leaves_matching_set
+                    )
+                    leaf_similarity_score = common_leaves_num / max(
+                        len(source_leaf_ids), len(target_leaf_ids)
+                    )
+
+                    adjusted_t = (
+                        self.t
+                        if min(len(source_leaf_ids), len(target_leaf_ids)) > 4
+                        else 0.4
+                    )
+
+                    if (
+                        similarity_score >= self.f
+                        and leaf_similarity_score >= adjusted_t
+                    ) or leaf_similarity_score >= 0.8:
+                        matching_set.add((source_node_id, target_node_id))
+                        self._unmatched_source_nodes.remove(source_node_id)
+                        self._unmatched_target_nodes.remove(target_node_id)
+                        ordered_unmatched_target_nodes.pop(target_node_id, None)
+                        break
+
+        return matching_set
+
+    def _compute_leaf_matching_set(self):
+        leaf_matchings = []
+        for source_leaf in _get_leaves(self._source):
+            for target_leaf in _get_leaves(self._target):
+                if _is_same_label(source_leaf, target_leaf):
```

Comment:
what is same_label, should it just be called same_type?

Reply:
Renamed